src/defargs.l

```diff
@@ -500,9 +501,25 @@ CPPC  "/\/"
 <CopyArgRound,CopyArgRound2,CopyArgSharp,CopyArgCurly>.  {
 					  *yyextra->copyArgValue += *yytext;
 					}
-<ReadTypeConstraint>[,)>]               {
+<ReadTypeConstraint>[,)]               {
                                           unput(*yytext);
                                           BEGIN(yyextra->lastExtendsContext);
+                                       }
+<ReadTypeConstraint>">"                {
+                                          if (yyextra->argSharpCount > 0)
+                                          {
+                                            yyextra->curTypeConstraint+=yytext;
+                                            --yyextra->argSharpCount;
+                                          }
+                                          else
+                                          {
+                                            unput(*yytext);
+                                            BEGIN(yyextra->lastExtendsContext);
+                                          }
+                                        }
+<ReadTypeConstraint>"<"                 {
+                                          yyextra->curTypeConstraint+=yytext;
+                                          ++yyextra->argSharpCount;
```

Comment:
What happens when the uses has an error in the number of `<` `>` pairs e.g. maore `>` than `<` (not a valid program but still ...)?

Reply:
What behavior would you recommend in this case? Note that the status quo is rather random behavior for examples described in the issue. I guess the behavior of doxygen on invalid programs is in general undefined. However I admit that we may come up with a friendly behavior for this particular type of syntax issue. E.g. raise a warning if a premature `[(){}]` is encountered (notice that the entrance to `<ReadTypeConstraint>` is already guarded as Java-only)?

Reply:
Actually I'm a bit confused as I see that `ReadTypeConstraint` is entered as soon as the rule `<ReadFuncArgType,ReadFuncArgPtr>"extends"` is satisfied (for Java) and the count is increased as soon as `<` is seen and decreased when `>` is seen, but the count is at that moment still > 0 and when no `,` or `)` is found the user remains in the `ReadTypeConstraint` and will leave it as soon as it sees a `,` or `)` but what will happen when the user is in a `<` ..`>` part and sees a `,` or `)` or is this not possible?

Reply:
(for reference: discussion on this continued in #8495)